test/unit/codegen/codegen_coreneuron_cpp_visitor.cpp

```diff
@@ -36,17 +36,18 @@ std::shared_ptr<CodegenCoreneuronCppVisitor> create_coreneuron_cpp_visitor(
     const std::shared_ptr<ast::Program>& ast,
     const std::string& /* text */,
     std::stringstream& ss) {
+    ImplicitArgumentVisitor().visit_program(*ast);
```

Comment:
Why do we need this one and the `PerfVisitor` below (and why didn't we need it before)?

Reply:
We fused all the "create code" functions into a single one: run coreneuron codegen. Before there used to be variants, one for printing only the instance struct, etc. Those had different visitors applied to the ast. One of them was ImplicitArgumentVisitor.

The issue is that half the CodegenCoreneuronCppVisitor API became public for testing purposes only. To avoid that, we just always run the full thing and check if we find the substring. Since now the substring can be found indented, we also need the ability to shift snippets to the required indent level.

Reply:
Understood, thanks for clarifying!